backend/app/views.py

```diff
@@ -112,24 +114,129 @@ def search_photographers(request):
-    is_searching_by_name = name is not None and name.strip() != ""
-    if is_searching_by_name:
-        matching_photographers = (Photographer.objects.filter(name__icontains=name)
-                                                      .order_by("name")
-                                                      .prefetch_related("photo_set"))
+    location = request.GET.get("location", None)
+    map_square = request.GET.get("square", None)
+    name_start = request.GET.get("name_starts_with", None)
+    order_by = request.GET.get("order_by", None)
+
+    # Pagination params
+    page_number = request.GET.get("page", None)
+    count_per_page = 50 
+
+    search_params = {}
+    
+    # Validating and adding all of the params
+    if name is not None and name.strip() != "":
+        search_params["name__icontains"] = name
+    if location is not None and location.strip() != "" and location in locations:
+        search_params["approx_loc"] = location 
+    if map_square is not None and map_square.strip() != "":
+        try:
+            map_square = int(map_square)
+            if map_square in squares:
+                search_params["map_square"] = map_square 
+        except TypeError:
+            pass
+    
+    # Planning to check for multiple name starts for this field 
+    # Implmenetaiton example in this stackoverflow entry 
+    #  (https://stackoverflow.com/questions/5783588/django-filter-on-same-option-with-multiple-possibilities)
+    if name_start is not None and name_start.strip() != "" and name_start in nameStartsWith:
+        search_params["name__istartswith"] = name_start 
+    print(search_params)
+
+    order_by_field = parse_order_by(order_by)
+
+    if len(search_params) == 0:
+        matching_photographers = Photographer.objects.all()
     else:
-        matching_photographers = (Photographer.objects.all()
-                                                      .order_by("name")
-                                                      .prefetch_related("photo_set")[:50])
+        matching_photographers = Photographer.objects.filter(**search_params)
+
+    matching_photographers.prefetch_related("photo_set")
+
+    if order_by_field is not None:
+        matching_photographers = matching_photographers.order_by(order_by_field)
+    photographers_paginator = Paginator(matching_photographers, count_per_page)
+    current_page = photographers_paginator.get_page(page_number)
 
     serialized_photographers = (
-        PhotographerSearchSerializer(matching_photographers, many=True)
+        PhotographerSearchSerializer(current_page.object_list, many=True)
     ) # add pagination here
-    res = Response(serialized_photographers.data)
+    res = Response({
+        "page_number": page_number,
+        "results": serialized_photographers.data,
+        "is_last_page": not current_page.has_next()
+    })
     return res
 
 
+@api_view(['GET'])
+def get_search_photographers_dropdown_options(request):
```

Comment:
For now, this is good. Ultimately, since the list of locations is static and not going to change, this is something that we might want to remove from code, turn into data, and save in the frontend to avoid an additional backend query to make the photographers page.